### ISSUE TITLE
Java execution with conditional and json filter fails

### ISSUE THREAD

**Issue by simmel** [elastic/logstash#9292]
And it works as it should with Ruby exec.
Use same config as in #9255 @original-brownbear 

```
[2018-03-27T11:00:58,730][DEBUG][logstash.filters.grok    ] Running grok filter {:event=>#<LogStash::Event:0x60c970d6>}
[2018-03-27T11:00:58,736][DEBUG][logstash.util.decorators ] filters/LogStash::Filters::Grok: adding tag {"tag"=>"syslog"}
[2018-03-27T11:00:58,738][DEBUG][logstash.filters.grok    ] Event now:  {:event=>#<LogStash::Event:0x60c970d6>}
[2018-03-27T11:00:58,748][DEBUG][logstash.filters.mutate  ] filters/LogStash::Filters::Mutate: removing field {:field=>"host"}
[2018-03-27T11:00:58,757][DEBUG][logstash.filters.mutate  ] filters/LogStash::Filters::Mutate: removing field {:field=>"port"}
[2018-03-27T11:00:58,758][DEBUG][logstash.filters.json    ] Running json filter {:event=>#<LogStash::Event:0x60c970d6>}
[2018-03-27T11:00:58,761][DEBUG][logstash.filters.json    ] Event after json filter {:event=>#<LogStash::Event:0x60c970d6>}
[2018-03-27T11:00:58,762][DEBUG][logstash.filters.json    ] Running json filter {:event=>#<LogStash::Event:0x60c970d6>}
[2018-03-27T11:00:58,765][WARN ][logstash.filters.json    ] Error parsing json {:source=>"message", :raw=>"some json", :exception=>#<LogStash::Json::ParserError: Unrecognized token 'some': was expecting ('true', 'false' or 'null')
 at [Source: (byte[])"some json"; line: 1, column: 6]>}
[2018-03-27T11:00:58,770][DEBUG][logstash.filters.fingerprint] String built {:to_checksum=>"|@timestamp|2018-03-27T08:59:41.973Z|logsource|host.domain.tld|message|some json|"}
[2018-03-27T11:00:58,773][DEBUG][logstash.filters.grok    ] Running grok filter {:event=>#<LogStash::Event:0x60c970d6>}
[2018-03-27T11:00:58,773][DEBUG][logstash.filters.grok    ] Event now:  {:event=>#<LogStash::Event:0x60c970d6>}
[2018-03-27T11:00:58,778][DEBUG][logstash.util.decorators ] filters/LogStash::Filters::Mutate: adding value to field {"field"=>"[@metadata][logsource_graph]", "value"=>["%{logsource}"]}
{
       "@timestamp" => 2018-03-27T08:59:41.973Z,
        "@metadata" => {
        "logsource_graph" => "host_domain_tld",
             "ip_address" => "127.0.0.1",
            "fingerprint" => "3cc5d61317f22fa2628cf43517f26a62"
    },
         "@version" => "1",
    "timestamp8601" => "2018-03-27T08:59:41.973Z",
          "program" => "program",
        "logsource" => "host.domain.tld",
          "message" => "some json",
             "tags" => [
        [0] "syslog",
        [1] "_jsonparsefailure"
    ]
}
```

For input you can use:
```
echo '2018-03-27T08:59:41.973Z host.domain.tld program: {"message": "some json"}' | nc 127.0.0.1 2514
```

**Comment by original-brownbear** [comment]
I think this may not be directly related to the Java execution but just be a side effect of using a non-UTF8 charset here.

```
input {
  tcp {
    host => "localhost"
    port => 2514
    codec => line {
        charset => "ISO8859-1"
    }
  }
}
```

is the input in the config used in #9255.

We have a number of other issues that suggest there may be a problem in our conversion between Java and Ruby `String`. Link #9281 and related issues.

This is pretty valuable in tracking those down, thanks :)

**Comment by original-brownbear** [comment]
@andrewvc fyi, getting closer to resolving this :)

**Comment by original-brownbear** [comment]
@simmel would you be able to share a few lines of input (just the raw lines that arrive at the tcp socket) so I can reproduce the charset issue more quickly?

**Comment by simmel** [comment]
I get the problem with the data I submitted when creating the issue. Don't you?

**Comment by simmel** [comment]
Doesn't have to *be* any latin1, input tcp just needs to be configured for it. The exception goes away when using UTF-8 instead of latin1.

**Comment by original-brownbear** [comment]
@simmel oh got it :) Thanks so much, this should be easy to track down now!

**Comment by original-brownbear** [comment]
@simmel fixed in #9294 :) and merged to `master` and `6.x`
